src/shared/Table/ResponsiveTable.tsx

```diff
@@ -68,7 +68,7 @@ export type ResponsiveTableProps<TRow, TCol> = {
   setRowOuiaId?: (props: RowProps<TRow>) => string;
   tableOuiaId?: string;
   variant?: TableVariant;
-  onCheck?: (isSelecting: boolean, rowIndex: number) => void;
+  onCheck?: (isSelecting: boolean, row: RowProps<TRow>) => void;
```

Comment:
The row param is always the first one in the other callbacks. I'd do the same here for consistency